components/aws/sagemaker/common/_utils.py

```diff
@@ -213,9 +215,9 @@ def get_model_artifacts_from_job(client, job_name):
 
 def get_image_from_job(client, job_name):
     info = client.describe_training_job(TrainingJobName=job_name)
-    try:
+    if 'TrainingImage' in info['AlgorithmSpecification']:
```

Comment:
I think we still need exceptional handling code, in case if info is null

Reply:
`info` cannot be `None`. If the training job does not exist, the `client.describe_training_job` command throws an exception. This case was not being handled before.